ci-operator/step-registry/cucushift/upgrade/retarget/cucushift-upgrade-retarget-commands.sh

```diff
@@ -0,0 +1,618 @@
+            && [[ "$(echo "$output" | jq -r '.status.conditions[] | select(.type == "ReleaseAccepted").status')" == "True" ]]; then
+            
+            if [[ "$TARGET_MINOR_VERSION" -gt 18 ]]; then
+                if [[ "$(echo "$output" | jq -r '.status.conditions[] | select(.type == "Upgradeable").status')" != "False" ]]; then
+                    echo "The upgrade has not started yet, retry ${retry}"
+                    retry=$(( retry+1 ))
+                    sleep 1m
+                    continue
+                fi
+            fi
+            
+            echo "Upgrade is processing"
+            return 0
+        fi
+        echo "The upgrade has not started yet, retry ${retry}"
+        retry=$(( retry+1 ))
+        sleep 1m
+    done
+    echo "Error: upgrade not started, break the job"
+    run_command "oc adm upgrade"
+    return 1
+}
+
+# Upgrade the cluster to target release
+function upgrade() {
+    # As https://issues.redhat.com/browse/OTA-861 required, we can re-target to a z version when an upgrade is processing.
+    # For example, there is one upgrade is in progress A -> B (no matter it is a y stream or a z stream upgrade)
+    # Then:
+    # 1. we can retarget to a version which has same minor version with B
+    # 2. we can NOT retarget to a version which minor version is great than B
+    # *******************
+    # Nightly build is not applicable for z stream retarget tests
+    # *******************
+    # No matter y stream or z stream retarget, the TARGET always the final version we want to upgrade to
+    # To make OCP-25473 not block the whole pipeline, the upgrade orders are:
+    #  if the minor version in ${SHARED_DIR}/upgrade-edge great than target, then
+    #       initial -> target -> ${SHARED_DIR}/upgrade-edge
+    #  if the minor version in ${SHARED_DIR}/upgrade-edge equal to target, then
+    #       1. we find a target+1 version as third target upgrade version
+    #       2. initial -> ${SHARED_DIR}/upgrade-edge -> target -> third target version
+    #  this case not allow ${SHARED_DIR}/upgrade-edge less than target
+
+    # INTERMEDIATE_MINOR_VERSIONT is the minor version in ${SHARED_DIR}/upgrade-edge
+    local retry=0 intermediate_image INTERMEDIATE_MINOR_VERSIONT FIRST_MINOR_VERSION SECOND_MINOR_VERSION block_second first_upgrade_to_image second_upgrade_to_image third_upgrade_to_image output latest_minor_version
+    local testcase="retarget"
+    export IMPLICIT_ENABLED_CASES="${IMPLICIT_ENABLED_CASES} ${testcase}"
+    
+    intermediate_image="$(< "${SHARED_DIR}/upgrade-edge")"
+    if [[ -z "${intermediate_image:-}" ]]; then
+        echo "Error: The intermediate image is not given, break the job"
+        return 1
+    fi
+    
+    echo "Prepare test data"
+    INTERMEDIATE_MINOR_VERSIONT="$(oc adm release info -ojson "$intermediate_image" | jq -r '.metadata.version' | cut -f2 -d.)"
```

Comment:
`INTERMEDIATE_MINOR_VERSIONT` did you mean `INTERMEDIATE_MINOR_VERSION`, noticed a `T` behind the variable

Reply:
Yes, it's a typo.